src/main/java/HttpRequestWrapper.java

```diff
@@ -0,0 +1,160 @@
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class HttpRequestWrapper {
+
+    private final HttpClient client;
+
+    public HttpRequestWrapper() {
+        this.client = HttpClient.newHttpClient();
+    }
+
+    public HttpResultData sendGet(String url) {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(url))
+                .GET()
```

Comment:
```suggestion
                .uri(URI.create(url))
                .timeout(timeout)
                .GET()
```
You should add timeout on the request level. Without it, hanging connections can eat up time. This is not the only place I think.